src/cmd/ksh93/sh/xec.c

```diff
@@ -1774,7 +1774,8 @@ int sh_exec(Shell_t *shp, const Shnode_t *t, int flags) {
                     struct checkpt *buffp =
                         (struct checkpt *)stkalloc(shp->stk, sizeof(struct checkpt));
                     shp->st.otrapcom = 0;
-                    if ((nsig = shp->st.trapmax * sizeof(char *)) > 0 || shp->st.trapcom[0]) {
+                    nsig = shp->st.trapmax * sizeof(char *);
```

Comment:
As you did in *subshell.c* the multiplication by `sizeof(char *)` should be deferred to the actual malloc call.